addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -73,13 +73,20 @@ class ComposerTextInput extends Component {
         if (!this.composer) {
             return "";
         }
-        if (this.composer.thread && this.composer.thread.model !== 'mail.channel') {
-            if (this.composer.isLog) {
-                return this.env._t("Log an internal note...");
+        if (this.composer.thread) {
+            if (this.composer.thread.model !== 'mail.channel') {
+                if (this.composer.isLog) {
+                    return this.env._t("Log an internal note...");
+                }
+                return this.env._t("Send a message to followers...");
+            } else {
+                const name = this.composer.thread.displayName;
+                if (this.composer.thread.channel_type !== 'chat') {
+                    return _.str.sprintf(this.env._t("Message #%s..."), name);
+                }
+                return _.str.sprintf(this.env._t("Message %s..."), name);
```

Comment:
nested conditions are hard to read.

Prefer a flat hierarchy with quick returns.

If there is no thread do a `return ""` (this is not a realistic case and we will soon remove this condition with required fields). That's already 1 less indent level.

Then I would do all the specific conditions for `mail.channel` also with returns.

And at the bottom the conditions for "not channel" (without nesting them in an `else` condition, that's not necessary if all the other conditions also do `return`).